File: 26.remove-duplicates-from-sorted-array.java
/*
 * @lc app=leetcode id=26 lang=java
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution {
    public int removeDuplicates(int[] nums) {
        int i = 1; // the 0th element always counts, i is the insertion index
        for(int j = 1; j < nums.length; j++){
            // how did we ensure that the new found element is unique from all previous elements?
            // it says the nums is non decreasing, meaning if some elements
            // are non unique, all duplicate values will be one after the other, 
            // it will not appear out of order
            if(nums[j] != nums[i-1]){ // nums[i-1] will be the last unique element
                nums[i] = nums[j];
                i++;
            }
        }
        return i;
    }
}
// @lc code=end

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.py
# Last updated: 9/14/2025, 12:55:53 PM
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        ue = nums[0]
        a = nums
        #front = 0
        k = 1 # number of unique element
        for i, num in enumerate(nums[1:]):
            if num != ue:
                ue = num
                nums[k] = num
                k += 1

            else:
                continue
        return k 